src/main/java/fi/aalto/cs/apluscourses/intellij/model/IntelliJCourse.java

```diff
@@ -1,23 +1,39 @@
 package fi.aalto.cs.apluscourses.intellij.model;
 
 import com.intellij.openapi.project.Project;
+import fi.aalto.cs.apluscourses.model.Component;
 import fi.aalto.cs.apluscourses.model.Course;
+import fi.aalto.cs.apluscourses.model.Library;
 import fi.aalto.cs.apluscourses.model.Module;
+import fi.aalto.cs.apluscourses.model.NoSuchModuleException;
 import java.util.List;
 import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
 import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
 
 class IntelliJCourse extends Course {
   @NotNull
   private final Project project;
+  @NotNull
+  private final Map<String, IntelliJModule> modulesByNames;
+  @NotNull
+  private final CommonLibraryProvider commonLibraryProvider;
 
-  IntelliJCourse(@NotNull String name,
-                 @NotNull List<Module> modules,
-                 @NotNull Map<String, String> requiredPlugins,
-                 @NotNull Project project) {
-    super(name, modules, requiredPlugins);
+  public IntelliJCourse(@NotNull String name,
+                        @NotNull List<Module> modules,
+                        @NotNull List<Library> libraries,
+                        @NotNull Map<String, String> requiredPlugins,
+                        @NotNull Project project) {
+    super(name, modules, libraries, requiredPlugins);
 
     this.project = project;
+    this.commonLibraryProvider = new CommonLibraryProvider(project);
+    this.modulesByNames = modules
+        .stream()
+        .map(IntelliJModule.class::cast)
+        .collect(Collectors.toMap(IntelliJModule::getName, Function.identity()));
```

Comment:
a note for myself: [Function._identity()_](https://docs.oracle.com/javase/8/docs/api/java/util/function/Function.html#identity--)